gcp_variant_transforms/libs/variant_partition.py

```diff
@@ -34,59 +44,266 @@
+
+    if config_file_path:
+      self._config_file_path_given = True
+      self._parse_config(config_file_path)
+
+  def _validate_config(self, config_file_path):
+    # type: (str) -> None
+    with FileSystems.open(config_file_path, 'r') as f:
+      try:
+        partition_configs = yaml.load(f)
+      except yaml.YAMLError as e:
+        raise ValueError('Invalid yaml file: %s' % str(e))
+    if len(partition_configs) > _MAX_NUM_PARTITIONS:
+      raise ValueError(
+          'There can be at most {} partitions but given config file '
+          'contains {}'.format(_MAX_NUM_PARTITIONS, len(partition_configs)))
+    if not partition_configs:
+      raise ValueError('There must be at least one partition in config file.')
+
+    existing_partition_names = set()
+    for partition_config in partition_configs:
+      partition = partition_config.get('partition', None)
+      if partition is None:
+        raise ValueError('Wrong yaml file format, partition field missing.')
+      regions = partition.get('regions', None)
+      if regions is None:
+        raise ValueError('Each partition must have at least one region.')
+      if len(regions) > _MAX_NUM_REGIONS:
+        raise ValueError('At most {} regions per partition, thie partition '
+                         'contains {}'.format(_MAX_NUM_REGIONS, len(regions)))
+      # Check the partition_name is unique.
+      if not partition.get('partition_name', None):
+        raise ValueError('Each partition must have partition_name field.')
+      partition_name = partition.get('partition_name').strip()
+      if partition_name == '':
+        raise ValueError('Partition name can not be empty string.')
+      if partition_name in existing_partition_names:
+        raise ValueError('Table names must be unique, '
+                         '{} is duplicated'.format(partition_name))
+      existing_partition_names.add(partition_name)
+    return partition_configs
+
+  def _parse_config(self, config_file_path):
+    # type: (str) -> None
+    """Parses the given partition config file.
+
+    Args:
+      config_file_path: name of the input partition_config file.
+    Raises:
+      A ValueError if any of the expected config formats are violated.
+    """
+    def _parse_position(pos_str):
+      # type: (str) -> int
+      return int(pos_str.replace(',', ''))
+
+    def _is_residual_partition(regions):
+      # type: (List[str]) -> bool
+      return (len(regions) == 1 and
+              regions[0].strip().lower() == _RESIDUAL_REGION_LITERAL)
+
+    partition_configs = self._validate_config(config_file_path)
+
+    self._num_partitions = len(partition_configs)
+    for partition_index in range(self._num_partitions):
+      partition = partition_configs[partition_index].get('partition')
+      self._partition_names[partition_index] = (
+          partition.get('partition_name').strip())
+      regions = partition.get('regions', None)
+
+      # Check whether this is the residual partition.
```

Comment:
nit: by changing that logic to a method, you no longer need this comment (one of the points of making this into a method), so please remove.